src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala

```diff
@@ -234,6 +234,7 @@ class TSDefParser extends StdTokenParsers with ImplicitConversions {
     | thisType
     | indexTypeQuery
     | "(" ~> typeDesc <~ ")"
+    | "|" ~> unionTypeDesc
```

Comment:
I am concerned that the TypeScript grammar in the specification ([here](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#A.1)) does not mention leading `|` in union types. This casts doubts on the precedence of that operator. However, I would guess that it should be inside `unionTypeDesc`, so that it does not modify the precedence of union types in general with respect to intersection types and base types.

So, concretely, I think it should be `opt("|") ~>` at the beginning of `unionTypeDesc`.

Reply:
That's fair. I'll try to make the change next week when I'm back from vacation.

I looked at that spec when I was trying to figure out what was going on with the leading `|`s, and I came to the conclusion that the spec is out of date. Do you have any idea whether that's true or not? (I agree that the spec doesn't mention leading `|`s but I've seen them in the TypeScript definitions Amazon's JavaScript SDK, IIRC, and they work in the [TypeScript Playground](http://www.typescriptlang.org/play/).)

Reply:
Yes, absolutely. The spec seems out of date.

Reply:
@sjrd FYI, I updated the parser in eb0f33f per your suggestion.